### PR TITLE
CORTX-33986 : Convert HLD for Data Integrity in .md format.

### PR DESCRIPTION
**Pull request by rajatpatil98**
# Problem Statement
- Convert existing Motr HLD in confluence https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/1010663429/Motr+DI in .md format and add to cortx-motr repo in github.
# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered doc/HLD-Data-Integrity.md](https://github.com/rajatpatil98/cortx-motr/blob/HLD_For_Di/doc/HLD-Data-Integrity.md)

### CODE REVIEW

**Review by vidyadhar-pinglikar — APPROVED** [review]

File: doc/HLD-Data-Integrity.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+- I. Motr Client
+  - I.1 Application and motr data structure
+  - I.2 Parity Group Computation 
+  - I.3 Tracking Data Unit Allocated to Object
+  - I.4 FOP Sending and Checksum Processing
+    - Write Path
+    - Read Path
+- II. Motr Server Write Path
+  - II.1 GOB => COB
```
Comment (rkothiya) [inline doc/HLD-Data-Integrity.md:11]: I would recommend to use the full form here

File: doc/HLD-Data-Integrity.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+- Balloc Extent Start
+
+  - ee_val
+
+![image](./Images/DI09.png)
+### II.4 COB-EMAP Details
+- When COB is created a default entry for the object extent is created
+
+  - Dummy hole (value 0) extent with a span of 0 to ∞
```
Comment (rkothiya) [inline doc/HLD-Data-Integrity.md:130]: You might want to change "Dummy" to "Fake"

File: doc/HLD-Data-Integrity.md (hunk 1 above)
Comment (rajatpatil98) [reply doc/HLD-Data-Integrity.md]: done

File: doc/HLD-Data-Integrity.md (hunk 2 above)
Comment (rajatpatil98) [reply doc/HLD-Data-Integrity.md]: done, used "Fake Extent"

File: doc/HLD-Data-Integrity.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+![image](./Images/DI05.png)
+### Read Path
+During read path when the data is received from Motr Server, the checksum is computed and compared against received checksum
+
+
+ 
+![image](./Images/DI06.png)
+## II. Motr Server Write Path
+### II.1 Global Object => Component Object
+Every Motr object is identified by FID also known as Global Object FID and its stripe on devices are identified as Component Object FID.
```
Comment (shashank-parulekar) [inline doc/HLD-Data-Integrity.md:75]:
```suggestion
Every Motr object is identified by FID also known as Global Object FID and its Stripe Units on devices are identified as Component Object FID.
```

File: doc/HLD-Data-Integrity.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+- Balloc will attempt to allocate total size as one extent
+
+  - If one chunk is not available then multiple balloc extent can be allocated
+
+  - Currently more than one chunk will cause failure
+
+- In the diagram below it is shown that three balloc extents are getting allocated for two data DUs.
+
+### Balloc extent and buffer extent processing
+As part of balloc processing, server codes find the number of contiguous fragment using overlap of balloc-extent and buffer extent. Also data structure is populated to track this.
```
Comment (shashank-parulekar) [inline doc/HLD-Data-Integrity.md:99]:
```suggestion
As part of balloc processing, server code finds the number of contiguous fragment using overlap of balloc-extent and buffer extent. Also data structure is populated to track this.
```

File: doc/HLD-Data-Integrity.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+- m0_indexvec si_stob : Tracking balloc fragment
+
+![image](./Images/DI07.png)
+<p align="center">Balloc Processing and Fragment Computation</p>
+
+These balloc-extent along with its buffer from unit for Storage IO.  
+
+![image](./Images/DI08.png)
+### II.3 EMAP Extent Processing
+As part of EMPA extent processing contiguous fragment is computed using overlap of Object offset extent (COB Offset) and balloc extent. These EMAP fragment data are processed later and gets written to the device EMAP btree.
```
Comment (shashank-parulekar) [inline doc/HLD-Data-Integrity.md:112]:
```suggestion
As part of EMAP extent processing, contiguous fragment is computed using overlap of Object offset extent (COB Offset) and balloc extent. This EMAP fragment data is processed later and gets written to the device EMAP btree.
```

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

**Review by shashank-parulekar — APPROVED** [review]

**Comment by rkothiya** [comment]
Giving an exception for ci/cd regression and pre-merge job as it is documentation change.

**Pull request merged**